### PR TITLE
[docs] Clarify REST service / mempool separation in the life of a transaction doc

### PR DESCRIPTION
**Pull request by banool**
### Description
More context: https://aptos-org.slack.com/archives/C03F71PU3B5/p1681381120814929.

### Test Plan
Docs.

### CODE REVIEW

File: developer-docs-site/docs/guides/basics-life-of-txn.md [hunk 1]
```diff
@@ -157,12 +157,15 @@ A client submits a transaction to the REST service of an Aptos fullnode.
 
 ### 2. REST Service → Mempool
 
-The REST service forwards the transaction to a validator fullnode, which then sends it to validator node V<sub>X</sub>'s mempool. The mempool will accept the transaction T<sub>N</sub> only if the sequence number of T<sub>N</sub> is greater than or equal to the current sequence number of the sender's account (note that the transaction will not be passed to consensus until the sequence number matches the sequence number of the sender’s account).
+The REST service of the fullnode puts the transaction in its mempool.
```
Comment (bchocho) [inline developer-docs-site/docs/guides/basics-life-of-txn.md:160]: I'll move the acceptance part back to this section. If the transaction is old, mempool rejects it here, and the REST service will tell the user about it

**Review by bchocho — APPROVED** [review]
LGTM

File: developer-docs-site/docs/guides/basics-life-of-txn.md [hunk 2]
```diff
@@ -216,7 +219,7 @@ Mempool is a shared buffer that holds the transactions that are “waiting” to
 
 ### 1. REST Service → Mempool
 
-* After receiving a transaction from the client, the REST service proxies the transaction to a validator fullnode. The transaction is then sent to the validator node’s mempool.
+* After receiving a transaction from the client, the REST service sends the transaction to its own mempool, which then shares the transaction with the mempool of a validator fullnode. The mempool on the validator fullnode then shares the transaction with the mempool of a validator.
```
Comment (clay-aptos) [inline developer-docs-site/docs/guides/basics-life-of-txn.md:222]: This is likely out of the scope of this PR. Still, I want to learn and help:
Is the transaction processed in any way by the validator fullnode's mempool. If so, how? If not, why not send the transaction straight to the validator? What is the purpose of this proxy? Load balancing? Some checking?

We don't seem to describe the purpose of a validator fullnode here:
https://aptos.dev/nodes/validator-node/operator/node-requirements

And we don't even mention them here:
https://aptos.dev/concepts/validator-nodes/

 I am happy to help with updates to those pages once I have a greater understanding of VFNs.

**Review by clay-aptos — APPROVED** [review]
Thanks, Daniel. LGTM! I made some small changes for clarity and consistency and posed one question I will also raise internally.

**Pull request merged**